machine_learning/utils.py

```diff
@@ -0,0 +1,131 @@
+import re
+from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score, classification_report
+from sklearn.externals import joblib
+
+def clean_text(raw_text):
+    """
+    This function cleans text data by performing
+    various operations on it.
+    1. Remove all characters except letters (a-z)
+    2. Converts all text to lowercase
+
+    Parameters
+    ----------
+    raw_text : string
+
+    Returns
+    -------
+    meaningful_text : string
+    """
+
+    #Remove special characters
+    temp_text = re.sub('[^a-zA-Z]', ' ', raw_text)
+
+    #Convert all words to lower case and split into words
+    cleaned_text = temp_text.lower()
+
+    #Join the words together to create a space separated string
+    return cleaned_text
+
+
+def preprocess(vectorizer_func, text, labels, max_features, max_df=1.0, min_df=1):
```

Comment:
I agree they are thin wrappers. I found these to be a convenient option when I was learning and already had some sklearn experience. 
Since I was only dealing with text models in the beginning, it was really a matter of choice when it came to these functions. However, as we try out more models, I think it may make sense to have a few preprocessing functions. 
Do you think it's better that we remove them?

Reply:
It's a judgement call. As long as you are open to _maybe_ removing them in the future (and putting them into documentation or something) I would be OK with merging this now.

Reply:
Oh, absolutely! I think once we're sure about their usefulness, we can either keep them or put them somewhere else (documentation).